Update copy-ssh-public-key.md

nam20485 opened the pull request:
Add a tip and accompanying alternative command describing the error received when using the prescribed clip command in a PowerShell terminal and provide an alternative command that works correctly and doesn't produce the error.

(Windows 11 frequently uses the new Windows Terminal, which defaults to using the PowerShell command line shell. In the PowerShell command line, the '>' operator is reserved and does not pipe the output of a command, but instead causes an error. In this case, using the 'cat' command and then using the bar operator to pipe it's output to the 'clip' command succeeds without producing the error.)

<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to an issue) and what changes you've made, we can triage your pull request to the best possible team for review.
-->

### Why:

Closes: #25909 

<!-- If there's an existing issue for your change, please link to it above.
If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted: https://github.com/github/docs/issues/new/choose. -->

### What's being changed (if available, include any code snippets, screenshots, or gifs):

The clip command for Windows users to copy their SSH key to the clipboard for adding to their GitHub account. (Step 1 of the "[Adding a new SSH key to your account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account#adding-a-new-ssh-key-to-your-account)" section)

![image](https://github.com/github/docs/assets/1829396/5294063f-863b-4167-b7be-46a50689fb09)

<!-- Let us know what you are changing. Share anything that could provide the most context.
If you made changes to the `content` directory, a table will populate in a comment below with links to the preview and current production articles. -->

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

nam20485 committed:
Update copy-ssh-public-key.md

Add a tip describing the error received when using the prescribed clip command in a PowerShell terminal and provide an alternative command that works correctly and doesn't produce the error.

(Windows 11 frequently uses the new Windows Terminal, which defaults to using the PowerShell command line shell. In the PowerShell command line, the '>' operator is reserved and does not pipe the output of a command, but instead causes an error. In this case, using the 'cat' command and then using the bar operator to pipe it's output to the 'clip' command succeeds without producing the error.)

nohaaaaa18 commented:
Adfgjkk

nohaaaaa18 commented:
Wtuioppj is

nohaaaaa18 commented:
Seguin and the course

Eloelo420thc reviewed (commented):
Git

stevecat commented:
Hi @nam20485, thanks so much for opening this! ✨ 

Do you know if using the `|` pipe would actually work for all Windows users, even those using command prompt and older versions of Powershell? It looks like that should work and, if so, I think we should just replace the current Windows example. That would make this a lot simpler.

If not, or I've misunderstood, your changes still look good, I'd just suggest combining your `{% tip %}` block with the other one that's suggesting using `clip.exe`. There's a good example [in this article](https://github.com/github/docs/blob/main/content/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/unlinking-your-email-address-from-a-locked-account.md?plain=1#L15-L23). Let me know if you have any questions!

nam20485 commented:
>Do you know if using the | pipe would actually work for all Windows users, even those using command prompt and older versions of Powershell? It looks like that should work and, if so, I think we should just replace the current Windows example.

@stevecat Good question. I thought the same thing, but my testing shows that is not the case unfortunately. The new command I created **does not work** on Command Prompt (cmd.exe).

So I don't think we can replace the existing command and we have add the alternative command to the existing one.

I will make a commit that updates my PR to use the "notes" approach that you suggested.

nam20485 committed:
Merge branch 'github:main' into patch-1

nam20485 committed:
Merge branch 'github:main' into patch-1

nam20485 committed:
change use of two tips in a row to a "notes" section

As directed by @stevecat. Follows example from here: https://github.com/github/docs/blob/main/content/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/unlinking-your-email-address-from-a-locked-account.md?plain=1#L15-L23

Juancanico5 reviewed (commented):
> 
Duplicate of #

Obeda1995 commented:
حسنا صديقي يمكنك بالتعديل على هذه النقطه الذي كنت اعلق عليها اكتب جمله
مفيده الى اصدقائي والموجودين واكون ممنونك شكرا جزيلا لك صديقي

في الثلاثاء، ١٣ يونيو ٢٠٢٣, ٣:٣٨ ص Nathan Miller ***@***.***>
كتب:

> @stevecat <https://github.com/stevecat> I think its ready for your review
> again- I committed a change to use a Notes section with two bullet points
> instead of two tips in a row. Let me know if you need me to make any other
> changes.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/github/docs/pull/25911#issuecomment-1588295762>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7EAFFABIVT4BVVUAXJNM3LXK6Y73ANCNFSM6AAAAAAZB42XVM>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

stevecat commented:
Hi again @nam20485, that looks great, thank you! ✨ 

Unfortunately, our site isn't doing a very good job of rendering your changes:

![image](https://github.com/github/docs/assets/12534592/c5edfb64-4733-4f85-8a66-aab45ff3fdd2)

I don't think this is related to your changes. I can reproduce the raw markdown if I remove everything from the note block. It's likely a specific combination of indentation and reusables 😅 I've raised an issue with our engineering team and once this is resolved, I'll circle back here to merge your pull request.

gracepark commented on data/reusables/gpg/copy-ssh-public-key.md:
````diff
@@ -20,12 +20,20 @@
+   - With Windows Subsystem for Linux (WSL), you can use `clip.exe`. Otherwise if `clip` isn't working, you can locate the hidden `.ssh` folder, open the file in your favorite text editor, and copy it to your clipboard.
+   - On newer versions of Windows that use the Windows Terminal, or anywhere else that uses the PowerShell command line, you may receive a `ParseError` stating that `The '&lt;' operator is reserved for future use.` In this case, the following alternative `clip` command should be used:
+   
+      ```shell
+      $ cat ~/.ssh/id_{% ifversion ghae %}rsa{% else %}ed25519{% endif %}.pub | clip
+      # Copies the contents of the id_{% ifversion ghae %}rsa{% else %}ed25519{% endif %}.pub file to your clipboard
+      ```
````
I believe the indentation just needs to be moved to the left 3 spaces. It should be aligned with the bulleted points:

![Screenshot 2023-08-08 at 5 21 46 PM](https://github.com/github/docs/assets/22055162/249751e7-07b7-4051-8e8b-2f33a4921e46)

I believe that should fix the rendering issue!

nam20485 replied:
Great, thanks @gracepark! That is certainly an easy fix. 😄 

@stevecat I will make the changes and commit.

nam20485 replied:
@gracepark Mine on my fork looks a little different wrt to the indentation, i.e. the code inside the \``` shell fence is not indented, but the entire \``` shell code-fence is indented:

![image](https://github.com/github/docs/assets/1829396/f6ac30ef-20f7-430e-9e08-d91a9ad46520)

nam20485 replied:
So I committed a fix for the indentation of the entire /```shell code fence: b043740ecb57c93d66b3249018459a15419882cd

Please let me know if that is the correct fix or if you need me to make further changes.

nam20485 committed:
Merge branch 'github:main' into patch-1

nam20485 committed:
make indenting match bullet points to fix rendering problem #25911

guntrip committed:
Merge branch 'main' into patch-1

stevecat commented on data/reusables/gpg/copy-ssh-public-key.md:
````diff
@@ -20,12 +20,20 @@
+   
+   **Notes:**
+   
+   - With Windows Subsystem for Linux (WSL), you can use `clip.exe`. Otherwise if `clip` isn't working, you can locate the hidden `.ssh` folder, open the file in your favorite text editor, and copy it to your clipboard.
+   - On newer versions of Windows that use the Windows Terminal, or anywhere else that uses the PowerShell command line, you may receive a `ParseError` stating that `The '&lt;' operator is reserved for future use.` In this case, the following alternative `clip` command should be used:
+   
+   ```shell
+   $ cat ~/.ssh/id_{% ifversion ghae %}rsa{% else %}ed25519{% endif %}.pub | clip
+   # Copies the contents of the id_{% ifversion ghae %}rsa{% else %}ed25519{% endif %}.pub file to your clipboard
+   ```
````
````suggestion
   - With Windows Subsystem for Linux (WSL), you can use `clip.exe`. Otherwise if `clip` isn't working, you can locate the hidden `.ssh` folder, open the file in your favorite text editor, and copy it to your clipboard.
   - On newer versions of Windows that use the Windows Terminal, or anywhere else that uses the PowerShell command line, you may receive a `ParseError` stating that `The '&lt;' operator is reserved for future use.` In this case, the following alternative `clip` command should be used:
   
```shell
   $ cat ~/.ssh/id_{% ifversion ghae %}rsa{% else %}ed25519{% endif %}.pub | clip
   # Copies the contents of the id_{% ifversion ghae %}rsa{% else %}ed25519{% endif %}.pub file to your clipboard
```

````

Hmm, so this is still rendering strangely. I've recreated Grace's indentation in this change and I've also added a space between the end of the codeblock and the closing liquid tag, as that can cause things to render incorrectly. Fingers crossed!

![Screenshot 2023-08-11 at 12 04 33](https://github.com/github/docs/assets/12534592/aa9d5395-0d04-4e71-9fc0-b1a801477c23)

guntrip committed:
Update data/reusables/gpg/copy-ssh-public-key.md

stevecat commented:
That's still rendering strangely. Let me ask our engineering team to take another look.

gracepark commented on data/reusables/gpg/copy-ssh-public-key.md:
````diff
@@ -20,12 +20,21 @@
+   **Notes:**
+   
+   - With Windows Subsystem for Linux (WSL), you can use `clip.exe`. Otherwise if `clip` isn't working, you can locate the hidden `.ssh` folder, open the file in your favorite text editor, and copy it to your clipboard.
+   - On newer versions of Windows that use the Windows Terminal, or anywhere else that uses the PowerShell command line, you may receive a `ParseError` stating that `The '&lt;' operator is reserved for future use.` In this case, the following alternative `clip` command should be used:
+   
+```shell
+   $ cat ~/.ssh/id_{% ifversion ghae %}rsa{% else %}ed25519{% endif %}.pub | clip
+   # Copies the contents of the id_{% ifversion ghae %}rsa{% else %}ed25519{% endif %}.pub file to your clipboard
+```
 
-   {% tip %}
-
-   **Tip:** With Windows Subsystem for Linux (WSL), you can use `clip.exe`. Otherwise if `clip` isn't working, you can locate the hidden `.ssh` folder, open the file in your favorite text editor, and copy it to your clipboard.
-
-   {% endtip %}
+    
+   {% endnote %}     
+   
````
````suggestion

   {% note %}

   **Notes:**

   - With Windows Subsystem for Linux (WSL), you can use `clip.exe`. Otherwise if `clip` isn't working, you can locate the hidden `.ssh` folder, open the file in your favorite text editor, and copy it to your clipboard.
   - On newer versions of Windows that use the Windows Terminal, or anywhere else that uses the PowerShell command line, you may receive a `ParseError` stating that `The '&lt;' operator is reserved for future use.` In this case, the following alternative `clip` command should be used:

   ```shell
   $ cat ~/.ssh/id_{% ifversion ghae %}rsa{% else %}ed25519{% endif %}.pub | clip
   # Copies the contents of the id_{% ifversion ghae %}rsa{% else %}ed25519{% endif %}.pub file to your clipboard
   ```

   {% endnote %}

````

gracepark replied:
Try this instead. When I do that, I get this:

![Screenshot 2023-08-11 at 9 07 35 AM](https://github.com/github/docs/assets/22055162/120fe52b-cf95-4f4c-b1ad-ae26c5a91675)

guntrip committed:
Update copy-ssh-public-key.md

gracepark committed:
Update copy-ssh-public-key.md

gracepark commented:
Everything looks good!

https://docs-25911-1f56bc.preview.ghdocs.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account#adding-a-new-ssh-key-to-your-account

Thanks all for your patience!

stevecat commented:
Wonderful, thank you so much @gracepark!!

gracepark committed:
Update adding-a-new-ssh-key-to-your-github-account.md

guntrip committed:
fix misnumbered item

guntrip reviewed (approved):
We got there! Thanks so much for sticking with us, @nam20485 and for making these changes in the first place. I'll get this merged now :+1:

GaganHonor commented:
That was so great , You all are so so hard working and pros , I ... loved reading this discussion  ╰(*°▽°*)╯
congratulations @nam20485 sir 🌟 and @gracepark mam and @stevecat sir so amazing. 🌟🌟(❁´◡`❁) ... I am new and i wish i will be some nearby of this much knowledge one day .

nam20485 commented:
Cool! No worries @stevecat, I thought it went pretty smoothly. I'm really excited to have been able to make a contribution to this site. Thanks to you and @gracepark for helping me out with it.
